rust/protocol/src/crypto.rs

```diff
@@ -136,7 +166,7 @@ mod test {
             .expect("valid hex");
         let ptext = [0u8; 35];
 
-        let ctext = super::aes_256_ctr_encrypt(&ptext, &key)?;
+        let ctext = super::aes_256_ctr_encrypt(&ptext, &key.try_into().unwrap())?;
```

Comment:
We have a policy (enforced by Clippy, I thought?) to only use `expect` and never `unwrap`, forcing documentation of why it's safe.

Reply:
So that's true, but it appears that clippy will not enforce that for code that's only visible during testing. If I apply the following change, clippy correctly yells at me:
```diff
diff --git a/rust/protocol/src/crypto.rs b/rust/protocol/src/crypto.rs
index 89288eae3..d0e52062f 100644
--- a/rust/protocol/src/crypto.rs
+++ b/rust/protocol/src/crypto.rs
@@ -126,6 +126,11 @@ pub fn aes256_ctr_hmacsha256_decrypt(
     aes_256_ctr_decrypt(&ctext[..ptext_len], cipher_key)
 }

+pub fn asdf() {
+    let e: ::std::result::Result<(), String> = Err("asdf".to_string());
+    e.unwrap();
+}
+
 #[cfg(test)]
 mod test {
     use super::Result;
```

I will make the change to use `.expect()` here. We could probably figure out how to make clippy check test code too.